<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Declare Statements Style"
    >
    <standard>
    <![CDATA[
        Declare statements should, by default, be written without curly braces/alternative syntax.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Declare statement written without curly braces.">
        <![CDATA[
declare(strict_types=1);
        ]]>
        </code>
        <code title="Invalid: Declare statement written with curly braces or alternative syntax.">
        <![CDATA[
declare(encoding='ISO-8859-1', ticks=1) {
    // Code.
}

declare(encoding='ISO-8859-1', ticks=10):
    declare(ticks=1):
    // Code.
    enddeclare;
enddeclare;
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
        strict_types declaration directive mustn't be written using curly braces, or with alternative syntax.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: strict_types written without curly braces.">
        <![CDATA[
declare(strict_types=1);
        ]]>
        </code>
        <code title="Invalid: strict_types written using alternative syntax/curly braces.">
        <![CDATA[
declare(strict_types=1, ticks=1) {
   // Code.
}
        ]]>
        </code>
    </code_comparison>
</documentation>
